File: src/main/java/io/bindernews/thegrackle/cards/Flock.java
package io.bindernews.thegrackle.cards;

import com.megacrit.cardcrawl.actions.AbstractGameAction;
import com.megacrit.cardcrawl.actions.common.DamageAction;
import com.megacrit.cardcrawl.cards.DamageInfo;
import com.megacrit.cardcrawl.core.AbstractCreature;
import io.bindernews.bnsts.CardVariables;
import io.bindernews.thegrackle.actions.AddHitsAction;
import io.bindernews.thegrackle.cardmods.ExtraHitsMod;
import io.bindernews.thegrackle.variables.ExtraHitsVariable;
import lombok.val;
import org.jetbrains.annotations.NotNull;

public class Flock extends BaseCard implements ExtraHitsVariable.Mixin {
    public static final CardConfig C =
            new CardConfig("Flock", CardType.ATTACK, CardRarity.COMMON, CardTarget.ENEMY);
    static final CardVariables VARS = CardVariables.config(c -> {
        c.cost(1, -1);
        c.damage(2, 4);
        c.magic(1, -1);
        c.add(ExtraHitsVariable.inst, 1, 2);
        c.addModifier(new ExtraHitsMod());
    });

    public Flock() {
        super(C, VARS);
    }

    @Override
    public void apply(@NotNull AbstractCreature p, AbstractCreature m) {
        val fx = AbstractGameAction.AttackEffect.BLUNT_LIGHT;
        val hits = getExtraHits();
        for (int i = 0; i < hits; i++) {
            addToBot(new DamageAction(m, new DamageInfo(p, damage, damageTypeForTurn), fx));
        }
        addToBot(new AddHitsAction(this, magicNumber, AddHitsAction.getPlayerCardGroups()));
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/cards/Flock.kt
package io.bindernews.thegrackle.cards

import com.megacrit.cardcrawl.actions.AbstractGameAction.AttackEffect
import com.megacrit.cardcrawl.actions.common.DamageAction
import com.megacrit.cardcrawl.cards.DamageInfo
import com.megacrit.cardcrawl.core.AbstractCreature
import io.bindernews.bnsts.CardVariables
import io.bindernews.thegrackle.actions.AddHitsAction
import io.bindernews.thegrackle.cardmods.ExtraHitsMod
import io.bindernews.thegrackle.helper.extraHits
import io.bindernews.thegrackle.helper.hits

class Flock : BaseCard(C, VARS) {
    override fun apply(p: AbstractCreature, m: AbstractCreature?) {
        val fx = AttackEffect.BLUNT_LIGHT
        val hits = extraHits
        for (i in 0 until hits) {
            addToBot(DamageAction(m, DamageInfo(p, damage, damageTypeForTurn), fx))
        }
        addToBot(AddHitsAction(this, magicNumber, AddHitsAction.getPlayerCardGroups()))
    }

    companion object {
        @JvmField val C = CardConfig("Flock", CardType.ATTACK, CardRarity.COMMON, CardTarget.ENEMY)
        val VARS = CardVariables().apply {
            cost(1, -1)
            damage(2, 4)
            magic(1, -1)
            hits(1, 2)
            addModifier(ExtraHitsMod())
        }
    }
}